java/org/apache/catalina/filters/CsrfPreventionFilter.java

```diff
@@ -110,45 +285,70 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 
             HttpSession session = req.getSession(false);
 
+            String requestedPath = getRequestedPath(req);
             boolean skipNonceCheck = skipNonceCheck(req);
             NonceCache<String> nonceCache = null;
 
             if (!skipNonceCheck) {
                 String previousNonce = req.getParameter(nonceRequestParameterName);
 
                 if (previousNonce == null) {
-                    if (log.isDebugEnabled()) {
-                        log.debug("Rejecting request for " + getRequestedPath(req) + ", session " +
-                                (null == session ? "(none)" : session.getId()) +
-                                " with no CSRF nonce found in request");
-                    }
-
-                    res.sendError(getDenyStatus());
-                    return;
-                }
+                    if (enforce(req, requestedPath)) {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Rejecting request for " + getRequestedPath(req) + ", session " +
```

Comment:
Looks like quite a few repetitions of log.debug blocks with very similar content.  Might be a good opportunity for a private function?

Reply:
If you really think it needs it. I don't find the control-flow difficult to follow, here, with the log messages in there. I think further obfuscating the log messages will just generate more code.

Reply:
I personally think that the code is cleaner and easier to maintain when repetitive blocks are encapsulated in a function, but do I "really think it needs it"? Nahh - your code, your decision :)